### PR TITLE
Add simple wrapper for ratelimited calls, partial RSVP implementation

### LINKED ISSUE #3: pull answer data from rsvp endpoint
## Problem

When people rsvp for events they complete answers to questions, and we are not capturing this.

## Solution

* Capture RSVP data from meetup api, https://www.meetup.com/meetup_api/docs/:urlname/events/:event_id/rsvps/#list.

### Requirement 
* do this for each event id we have
*  capture answers
    * meetup api give us a list, let's store each answer separately
* also capture the flowing info for each answer
  * question
  * question id
  * member id
  * group id

### Gotcha
* answers and questions may not always exist

can't happen in parallel with #5 , one will block the other

### PR DESCRIPTION
**Pull request by gtungatkar**
closes #3

### CODE REVIEW

**Comment by gtungatkar** [comment]
@zassmin seems like some permission/authz issue. For the sample event id, I still can't see answers. The api [doc](https://www.meetup.com/meetup_api/docs/:urlname/events/:event_id/rsvps/#list) says this 

> Answers to event survey questions. Visible only to the posting member, hosts and the lead team

File: crawler/crawler_lib.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package crawler
+
+import (
+	"net/http"
+    "fmt"
+	"github.com/juju/ratelimit"
+    "strings"
+)
+/*
```
Comment (gtungatkar) [inline crawler/crawler_lib.go]: @derekdomo you should be able to leverage this framework for ratelimiting and pagination for group API calls.

File: crawler/rsvp_crawler.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	rsvpResp.rsvplist = dat.([]interface{})
+	return err
+}
+
+var apikey string = os.Getenv("MEETUP_API_KEY")
+
+func GetAnswers(event_id string) string {
+	fmt.Printf("get rsvp\n")
+	//generate api url
+	apiUrl := fmt.Sprintf("https://api.meetup.com/women-who-code-sf/events/%s/rsvps?key=%s", event_id, apikey)
```
Comment (zassmin) [inline crawler/rsvp_crawler.go]: This needs the `fields=answers` in order for it to return answers as part of the data. 

ex:

```
"https://api.meetup.com/women-who-code-sf/events/%s/rsvps?fields=answers&key=%s"
```

Comment (gtungatkar) [inline crawler/rsvp_crawler.go]: Dang, dumb of me. Thanks for pointing that out.

File: crawler/crawler_lib.go (hunk 1 above)
Comment (derekdomo) [inline crawler/crawler_lib.go]: Thanks, too busy last weekend. Will find some time tonight to fix it.

File: crawler/rsvp_crawler.go (hunk 2 above)
Comment (zassmin) [inline crawler/rsvp_crawler.go]: non of that!!!

image

File: crawler/rsvp_crawler.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	GROUP_ID    string
+}
+
+type RsvpResponse struct {
+	rsvplist    []interface{}
+	nextPageUrl string
+}
+
+/* Parses response to RSVP api and Satisfies the Entity interface */
+func (rsvpResp *RsvpResponse) ParseResponse(resp *http.Response) error {
```
Comment (derekdomo) [inline crawler/rsvp_crawler.go]: Super minor detail. Is this a internal used method or Will it be exported? If internally used, maybe make it lower case is better.

Comment (gtungatkar) [inline crawler/rsvp_crawler.go]: This is implementing the Entity interface in the crawler_lib. So this method has to be public.

File: crawler/rsvp_crawler.go (hunk 2 above)
Comment (gtungatkar) [inline crawler/rsvp_crawler.go]: Hi Zassmin, can you please take a look at the updated PR.

**Comment by zassmin** [comment]
@gtungatkar what issue does this close?

File: meetupgoers.go [hunk 4]
```diff
@@ -4,13 +4,15 @@ import (
...
 	"os"
-
 	"github.com/WomenWhoCode/meetupgoers/crawler"
 )
 
 func main() {
 	if len(os.Args) >= 2 && os.Args[1] == "scheduler" {
 		crawler.StartTheEngine()
+	} else if len(os.Args) >= 2 && os.Args[1] == "rsvp_test" {
+		event_id := "slmrkmywdbcc"
```
Comment (zassmin) [inline meetupgoers.go]: mental note: update this to pull events from the collection if there are any in https://github.com/WomenWhoCode/meetupgoers/issues/9

Comment (gtungatkar) [inline meetupgoers.go]: Thanks much.

**Pull request merged**